### PR TITLE
feat: add get smarter api client

### PR DESCRIPTION
**Pull request by long74100**
https://2u-internal.atlassian.net/browse/ENT-6090

Adds a client to interface with the get smarter api gateway.

# ⛔️ DEPRECATION WARNING

**This repository is deprecated and in maintainence-only operation while we work on a replacement, please see [this announcement](https://discuss.openedx.org/t/deprecation-removal-ecommerce-service-depr-22/6839) for more information.**

Although we have stopped integrating new contributions, we always appreciate security disclosures and patches sent to [security@edx.org](mailto:security@edx.org)

## Required Testing
- [ ] Before deploying this change, complete a purchase in the stage environment. 
(^ We can remove that manual check once REV-2624 is done and the corresponding e2e test runs again)

## Description

Describe what this pull request changes, and why these changes were made. How will these changes affect other people, installations of edx, etc.?
Please include links to any relevant ADRs, design artifacts, and decision documents. Make sure to document the rationale behind significant changes in the repo, per [OEP-19](https://open-edx-proposals.readthedocs.io/en/latest/oep-0019-bp-developer-documentation.html), and can be
linked here.

Useful information to include:
- Which edX user roles will this change impact? Common user roles are "Learner", "Course Author", "Developer", and 
- 
- "Operator".
- Include screenshots for changes to the UI (ideally, both "before" and "after" screenshots, if applicable).
- Provide links to the description of corresponding configuration changes. Remember to correctly annotate these changes.

## Supporting information

Link to other information about the change, such as Jira issues, GitHub issues, or Discourse discussions.
Be sure to check they are publicly readable, or if not, repeat the information here.

## Testing instructions

Please provide detailed step-by-step instructions for testing this change; how did YOU test this change?

## Other information

Include anything else that will help reviewers and consumers understand the change.
- Does this change depend on other changes elsewhere?
- Any special concerns or limitations? For example: deprecations, migrations, OpenEdx vs. edx.org differences, development vs. production environment differences, security, or accessibility.

### CODE REVIEW

File: ecommerce/extensions/api_client/get_smarter.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    """
+    Client to interface with the GetSmarter Enterprise API Gateway(GEAG).
+    """
+
+    def __init__(self):
+        required_settings = [
+            'GET_SMARTER_OAUTH2_KEY',
+            'GET_SMARTER_OAUTH2_SECRET',
+            'GET_SMARTER_OAUTH2_PROVIDER_URL',
+            'GET_SMARTER_API_URL'
```
Comment (adamstankiewicz) [inline ecommerce/extensions/api_client/get_smarter.py:25]: [clarification] Are these settings all configured already?

File: ecommerce/extensions/api_client/get_smarter.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        except Exception as ex:
+            logger.exception(ex)
+            return None
+
+    def _ensure_authentication(self):
+        """
+        Add the required headers for authentication.
+        """
+        headers = {
+            # 'User-Agent': 'Mozilla/5.0',  # GEAG blocks the python-requests user agent for certain requests
```
Comment (adamstankiewicz) [inline ecommerce/extensions/api_client/get_smarter.py]: [clarification] Is this intended to be commented out?

Comment (long74100) [reply ecommerce/extensions/api_client/get_smarter.py]: nope

File: ecommerce/extensions/api_client/get_smarter.py (hunk 1 above)
Comment (long74100) [reply ecommerce/extensions/api_client/get_smarter.py:25]: Not yet, to test locally you can put them in devstack.py.

**Review by adamstankiewicz — APPROVED** [review]

File: ecommerce/extensions/api_client/get_smarter.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+            'User-Agent': 'Mozilla/5.0',  # GEAG blocks the python-requests user agent for certain requests
+            'Authorization': 'Bearer ' + self._get_access_token(),
+        }
+        self.session.headers.update(headers)
+
+    def get_terms_and_conditions(self):
+        """
+        Fetch and return the terms and conditions from GEAG.
+        """
+        url = self.api_url + '/terms'
```
Comment (adamstankiewicz) [inline ecommerce/extensions/api_client/get_smarter.py:96]: nit: `f"{self.api_url}/terms"`

**Pull request closed without merging**